### PR TITLE
prometheus metrics support

### PR DESCRIPTION
**Pull request by dfengshi**
attempt at addressing https://github.com/banzaicloud/bank-vaults/issues/72 similar to the banzai-charts helm chart

@bonifaido Hello, as we discussed in the issue #72 , here's a quick attempt at addressing the issue with a sidecar statsd container.

Please feel free to give copious feedback/help as I'm new to the operator pattern and go in general. Apologies if there are any glaring issues, I'll do my best to address. I do have a few questions about the volume mount and the TLS Env.

Thanks!

### CODE REVIEW

**Comment by bonifaido** [comment]
The TLS volumes are needed since the Vault Api is exposed through self signed certificate, so the CA is needed.

**Comment by dfengshi** [comment]
@bonifaido thanks, my bad

**Comment by bonifaido** [comment]
Only the formatting is now what needs to be fixed (go fmt)

File: operator/pkg/stub/handler.go [hunk 1]
```diff
@@ -417,6 +417,28 @@ func statefulSetForVault(v *v1alpha1.Vault) (*appsv1.StatefulSet, error) {
...
+							Ports: []v1.ContainerPort{{
+								Name:          "statsd",
+								ContainerPort: 9125,
+								Protocol:      "UDP",
+							}, {
+								Name:          "prometheus",
+								ContainerPort: 9102,
+								Protocol:      "TCP",
+							}},
+							VolumeMounts: withStatsDVolumeMount(v, withCredentialsVolumeMount(v, []v1.VolumeMount{})),
```
Comment (bonifaido) [inline operator/pkg/stub/handler.go]: `withCredentialsVolumeMount` is not needed for statsd. It is used to access on-cloud resources such as S3 or KMS for example.

File: operator/pkg/stub/handler.go [hunk 2]
```diff
@@ -484,6 +506,14 @@ func withTLSVolumeMount(v *v1alpha1.Vault, volumeMounts []v1.VolumeMount) []v1.V
 	return volumeMounts
 }
 
+func withStatsDVolumeMount(v *v1alpha1.Vault, volumeMounts []v1.VolumeMount) []v1.VolumeMount {
```
Comment (bonifaido) [inline operator/pkg/stub/handler.go]: The `statsd-mapping` volume has to be created as in: https://github.com/banzaicloud/banzai-charts/blob/master/vault/templates/deployment.yaml#L178

**Comment by dfengshi** [comment]
@bonifaido Hi Nandor, I've added the method `withStatsDVolume()` with the configmap setup as requested, but I'm fuzzy about what's going on in `deploymentForConfigurer()` and `statefulSetForVault()` vs what's going on in the helm chart. Should I be calling `withStatsDVolume()` in both of those methods? Maybe something like this: `withStatsDVolume(v, withTLSVolume(v, withCredentialsVolume(v, []v1.Volume{` ? 

This might be a bad question, because I might be missing some things in terms of understanding.

I've also added you as a collaborator on my fork in case it might be easier for you to just go ahead and dive in to my broken code rather than wait for me to understand what's missing.

Thanks again for the guidance+patience!

**Comment by bonifaido** [comment]
Hi David, I have pushed a fix to your branch, if you don't mind I will rebase your commits into one, so two commits will arrive in the master branch. I have tested the operator and now it is working and exposing the Prometheus metrics correctly, thank you for your contribution! 👍

**Comment by dfengshi** [comment]
@bonifaido Thanks Nandor!

**Pull request closed without merging**